Client/src/Pages/Settings Pages/StudentSettings/ProfileSettings/ConfirmImageDialog.js

```diff
@@ -0,0 +1,133 @@
+        .then((response) => response.json())
+        .then((presignedPostData) => {
+            const formData = new FormData();
+            Object.keys(presignedPostData.fields).forEach((field) => {
+                formData.append(field, presignedPostData.fields[field])
+            })
+            formData.append("file", file);
+            fetch(presignedPostData.url, {
+                method: 'POST',
+                body: formData
+            }).
+            then((response) => window.location.reload(false))
+        })
+    }
+
+    convertToJPEG = (file) => {
+        return new Promise((resolve) => {
+            var canvas = document.createElement('canvas');
+            var ctx = canvas.getContext('2d');
+            var image = new Image();
+            image.onload = (e) => {
+                canvas.width = e.target.width;
+                canvas.height = e.target.height;
+                drawImage(e.target);
+            }
+            image.src = window.URL.createObjectURL(file);
+
+            function drawImage(image) {
+                var ctx = canvas.getContext('2d');
+                ctx.drawImage(image, 0, 0, canvas.width, canvas.height);
+
+                convertToJPEG();
+            }
+
+            var convertToJPEG = () => {
+                canvas.toBlob((blob) => {
+                    console.log(blob);
+                    resolve(blob);
+                }, 'image/jpeg', 1.0);
+            }
+        })
+    }
+
+    // saveBlobAs(blob, )
+
+    render() {
+        if (this.state.loading) {
+            return (
+                <div align="center">
+                <br></br>
+                <Typography variant="h4">Loading...</Typography>
+                <br></br>
+                <CircularProgress />
+                </div>
+            )
+        }
+
+        return (
+            <Dialog
+                open={this.state.open}
+                onClose={this.toggleOpen}
+                fullWidth
+                maxWidth={'md'}
+            >
+                <DialogTitle onClose={this.toggleOpen}>
+                    Change Profile Picture
+                    <IconButton onClick={this.toggleOpen} style={{position: 'absolute', right: '0px', top: '0px' }}>
+                        <CloseIcon />
+                    </IconButton>
+                </DialogTitle>
+                <DialogContent
+                    dividers
+                    style={{ overflow: 'hidden' }}
+                >
+                    <center>
+                        <img
+                            src={this.state.picture ? window.URL.createObjectURL(this.state.picture) : ""}
+                            style={{
+                                maxWidth: '100%',
+                                maxHeight: '70vh', // Didn't knew the way to do this relatively to Dialog, so used absolute values. (Dialog by default uses 80vh)
```

Comment:
@mattrheault do you know how to do this relative to the parent rather than using absolute values?

Reply:
@ishchhabra wouldn't we only want to set either height or width here and set the other to auto since we wont know the image dimensions? Unless we want to crop images to certain dimensions beforehand. It could even be useful to just use the Avatar component instead since that is what is displayed on the navbar, and then the user will know what it looks like when it gets cropped to a circle for the navbar. In an ideal world we could have that but where they could drag and resize, but that would need to be backlogged for now so we can fix bugs.

Reply:
We are setting the max values for both, not the absolute size. So, this prevents overflow in both directions.

Reply:
Ahhh gotcha, missed that